src/hud.cpp

```diff
@@ -574,10 +576,24 @@ void Hud::updateSelectionMesh(const v3s16 &camera_offset)
 		0,0,1,1
 	};
 
-	m_selection_mesh = convertNodeboxesToMesh(m_selection_boxes, texture_uv);
+	// Use single halo box instead of multiple overlapping boxes.
+	// Temporary solution - problem can be solved with multiple
+	// rendering targets, or some method to remove inner surfaces.
+	// Thats because of halo transparency.
+
+	aabb3f halo_box = aabb3f(100.0, 100.0, 100.0, -100.0, -100.0, -100.0);
+	m_halo_boxes.clear();
+
+	for (std::vector<aabb3f>::iterator
+			i = m_selection_boxes.begin();
+			i != m_selection_boxes.end(); ++i) {
+		aabb3f box = *i;
```

Comment:
then in next 30 lines we will have *i instead of box, that will make code unreadable